errors.go

```diff
@@ -0,0 +1,46 @@
+package containers
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+type ErrHandler interface {
```

Comment:
> This means you need to wrap every handler with a redirection error handler.

This pull request added a content of ErrHandler interface, any error that implements the ErrHandler interface, will handle it self.